services/121-service/src/transaction-queues/dto/safaricom-transaction-job.dto.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  IsBoolean,
+  IsNotEmpty,
+  IsNumber,
+  IsNumberString,
+  IsString,
+} from 'class-validator';
+
+export class SafaricomTransactionJobDto {
+  @IsNotEmpty()
+  @IsNumberString()
+  public readonly programId: number;
+
+  @IsNotEmpty()
+  @IsNumberString()
+  public readonly paymentNumber: number;
+
+  @IsNotEmpty()
+  public readonly referenceId: string;
+
+  @IsNotEmpty()
+  @IsNumber()
+  public readonly transactionAmount: number;
+
+  @IsNotEmpty()
+  @IsBoolean()
+  public readonly isRetry: boolean;
+
+  @IsNotEmpty()
+  @IsNumber()
+  public readonly userId: number;
+
+  @IsNotEmpty()
+  @IsNumber()
+  public readonly bulkSize: number;
+
+  @IsNotEmpty()
+  @IsString()
+  public readonly phoneNumber: string;
```

Comment:
Should be optional, since this could be undefined if the program is misconfigured. Then it should still create the job, and only create the failed transaction when processing the job.